### PR TITLE
Add scan type in webhook event

### LINKED ISSUE #20331: Scan failed event needs to be updated for supporting SBOM
We need to mention the type, like it is an sbom generation failure.

{"type":"SCANNING_FAILED","occur_at":1713857684,"operator":"admin","event_data":{"resources":[{"digest":"sha256:6cb561c92b5df3f5e399d9e30c8a8353d8b8012a90a50af26068dc2f54c390d0","resource_url":"10.206.88.128/redis/redis@sha256:6cb561c92b5df3f5e399d9e30c8a8353d8b8012a90a50af26068dc2f54c390d0","scan_overview":{"application/vnd.security.vulnerability.report; version=1.1":{"report_id":"4ca91558-f4a8-424c-a373-d1118c8a959b","scan_status":"Error","severity":"","duration":32,"summary":null,"start_time":"2024-04-23T07:26:16Z","end_time":"2024-04-23T07:26:48Z","complete_percent":0}}}],"repository":{"name":"redis","namespace":"redis","repo_full_name":"redis/redis","repo_type":"private"}}}

### PR DESCRIPTION
**Pull request by stonezdj**
fixes #20331

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [ ] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [ ] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Review by zyyw — APPROVED** [review]

File: src/pkg/notifier/model/event.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ type EventData struct {
 	Repository  *Repository        `json:"repository,omitempty"`
 	Replication *model.Replication `json:"replication,omitempty"`
 	Retention   *model.Retention   `json:"retention,omitempty"`
+	ScanType    string             `json:"scan_type,omitempty"`
```
Comment (chlins) [inline src/pkg/notifier/model/event.go:45]: The `EventData` is a common data structure for the event, the content inside might need to be abstracted into new structures according to modules, for the sake of future expansion, should we abstract a `Scan` struct and put `ScanType` inside it? Just like replication and retention.

Comment (stonezdj) [reply src/pkg/notifier/model/event.go]: Updated

**Review by wy65701436 — APPROVED** [review]
lgtm

**Review by chlins — APPROVED** [review]
lgtm

**Pull request merged**